From 4e678061d1c30d6da90c690e150353d48be417a5 Mon Sep 17 00:00:00 2001
From: Wenyou Yang <wenyou.yang@atmel.com>
Date: Mon, 19 Jan 2015 09:37:10 +0800
Subject: [PATCH] board: sama5d4: remove the unused code related with
 PCK/MCK=396/132MHz

In the current version, the PCK/MCK=396/132MHz is not used for the sama5d4 board, so remove.

Signed-off-by: Wenyou Yang <wenyou.yang@atmel.com>
---
 board/sama5d4_xplained/Config.in.board    |  1 -
 board/sama5d4_xplained/sama5d4_xplained.c | 31 +++----------------------------
 board/sama5d4_xplained/sama5d4_xplained.h | 28 +---------------------------
 board/sama5d4ek/Config.in.board           |  1 -
 board/sama5d4ek/sama5d4ek.c               | 31 +++----------------------------
 board/sama5d4ek/sama5d4ek.h               | 28 +---------------------------
 6 files changed, 8 insertions(+), 112 deletions(-)

diff --git a/board/sama5d4_xplained/Config.in.board b/board/sama5d4_xplained/Config.in.board
index 683facf..8944605 100644
--- a/board/sama5d4_xplained/Config.in.board
+++ b/board/sama5d4_xplained/Config.in.board
@@ -7,7 +7,6 @@ config CONFIG_SAMA5D4_XPLAINED
 	select ALLOW_NANDFLASH
 	select ALLOW_SDCARD
 	select ALLOW_HSMCI
-	select ALLOW_CPU_CLK_396MHZ
 	select ALLOW_CPU_CLK_444MHZ
 	select ALLOW_CPU_CLK_510MHZ
 	select ALLOW_CPU_CLK_528MHZ
diff --git a/board/sama5d4_xplained/sama5d4_xplained.c b/board/sama5d4_xplained/sama5d4_xplained.c
index f3d94c5..7dadb98 100644
--- a/board/sama5d4_xplained/sama5d4_xplained.c
+++ b/board/sama5d4_xplained/sama5d4_xplained.c
@@ -111,32 +111,7 @@ static void ddramc_reg_config(struct ddramc_register *ddramc_config)
 				| AT91C_DDRC2_UNAL_SUPPORTED);
 
 
-#if defined(CONFIG_BUS_SPEED_133MHZ)
-
-	ddramc_config->rtr = 0x208;
-
-	/* One clock cycle @ 133 MHz = 7.5 ns */
-	ddramc_config->t0pr = (AT91C_DDRC2_TRAS_(6)
-			| AT91C_DDRC2_TRCD_(2)
-			| AT91C_DDRC2_TWR_(2)
-			| AT91C_DDRC2_TRC_(8)
-			| AT91C_DDRC2_TRP_(2)
-			| AT91C_DDRC2_TRRD_(2)
-			| AT91C_DDRC2_TWTR_(2)
-			| AT91C_DDRC2_TMRD_(2));
-
-	ddramc_config->t1pr = (AT91C_DDRC2_TXP_(2)
-			| AT91C_DDRC2_TXSRD_(200)
-			| AT91C_DDRC2_TXSNR_(19)
-			| AT91C_DDRC2_TRFC_(17));
-
-	ddramc_config->t2pr = (AT91C_DDRC2_TFAW_(5)
-			| AT91C_DDRC2_TRTP_(2)
-			| AT91C_DDRC2_TRPA_(2)
-			| AT91C_DDRC2_TXARDS_(2)
-			| AT91C_DDRC2_TXARD_(8));
-
-#elif defined(CONFIG_BUS_SPEED_148MHZ)
+#if defined(CONFIG_BUS_SPEED_148MHZ)
 
 	ddramc_config->rtr = 0x243;
 
@@ -227,7 +202,7 @@ static void ddramc_init(void)
 	pmc_enable_system_clock(AT91C_PMC_DDR);
 
 	/* configure Shift Sampling Point of Data */
-#if defined(CONFIG_BUS_SPEED_133MHZ) || defined(CONFIG_BUS_SPEED_148MHZ)
+#if defined(CONFIG_BUS_SPEED_148MHZ)
 	reg = AT91C_MPDDRC_RD_DATA_PATH_NO_SHIFT;
 #else
 	reg = AT91C_MPDDRC_RD_DATA_PATH_ONE_CYCLES;
@@ -241,7 +216,7 @@ static void ddramc_init(void)
 	reg &= ~AT91C_MPDDRC_CALCODEP;
 	reg &= ~AT91C_MPDDRC_CALCODEN;
 	reg |= AT91C_MPDDRC_RDIV_DDR2_RZQ_50;
-#if defined(CONFIG_BUS_SPEED_133MHZ) || defined(CONFIG_BUS_SPEED_148MHZ)
+#if defined(CONFIG_BUS_SPEED_148MHZ)
 	reg |= AT91C_MPDDRC_TZQIO_4;	/* @ 133 & 148 MHz */
 #else
 	reg |= AT91C_MPDDRC_TZQIO_5;	/* @ 170 & 176 MHz */
diff --git a/board/sama5d4_xplained/sama5d4_xplained.h b/board/sama5d4_xplained/sama5d4_xplained.h
index 3e4280f..4106e39 100644
--- a/board/sama5d4_xplained/sama5d4_xplained.h
+++ b/board/sama5d4_xplained/sama5d4_xplained.h
@@ -38,33 +38,7 @@
 
 #define BOARD_MAINOSC		12000000
 
-#if defined(CONFIG_CPU_CLK_396MHZ)
-/* PCK: 396M, MCK: 132M */
-#define BOARD_PCK		((unsigned long)((BOARD_MAINOSC * 66) / 2))
-#define BOARD_MCK		((unsigned long)((BOARD_MAINOSC * 66) / 2 / 3))
-
-#define BOARD_CKGR_PLLA		(AT91C_CKGR_SRCA | AT91C_CKGR_OUTA_0)
-#define BOARD_PLLACOUNT		(AT91C_CKGR_PLLACOUNT && (0x3F << 8))
-#define BOARD_MULA		((AT91C_CKGR_MULA << 2) & (65 << 18))
-#define BOARD_DIVA		(AT91C_CKGR_DIVA & 1)
-
-/* Master Clock Register */
-#define BOARD_PRESCALER_MAIN_CLOCK	(AT91C_PMC_PLLADIV2_2 | \
-					AT91C_PMC_MDIV_3 | \
-					AT91C_PMC_CSS_MAIN_CLK)
-
-#define BOARD_PRESCALER_PLLA		(AT91C_PMC_PLLADIV2_2 | \
-					AT91C_PMC_MDIV_3 | \
-					AT91C_PMC_CSS_PLLA_CLK)
-
-#define BOARD_H32MX			(AT91C_PMC_H32MXDIV_H32MXDIV2 \
-					| AT91C_PMC_PLLADIV2_2 \
-					| AT91C_PMC_MDIV_3 \
-					| AT91C_PMC_CSS_PLLA_CLK)
-
-#define MASTER_CLOCK		132000000
-
-#elif defined(CONFIG_CPU_CLK_444MHZ)
+#if defined(CONFIG_CPU_CLK_444MHZ)
 
 /* PCK: 444M, MCK: 148M */
 #define BOARD_PLLA_MULA		73
diff --git a/board/sama5d4ek/Config.in.board b/board/sama5d4ek/Config.in.board
index b5e4b16..4611bcf 100644
--- a/board/sama5d4ek/Config.in.board
+++ b/board/sama5d4ek/Config.in.board
@@ -7,7 +7,6 @@ config CONFIG_SAMA5D4EK
 	select ALLOW_NANDFLASH
 	select ALLOW_SDCARD
 	select ALLOW_HSMCI
-	select ALLOW_CPU_CLK_396MHZ
 	select ALLOW_CPU_CLK_444MHZ
 	select ALLOW_CPU_CLK_510MHZ
 	select ALLOW_CPU_CLK_528MHZ
diff --git a/board/sama5d4ek/sama5d4ek.c b/board/sama5d4ek/sama5d4ek.c
index 95b34fc..754f43c 100644
--- a/board/sama5d4ek/sama5d4ek.c
+++ b/board/sama5d4ek/sama5d4ek.c
@@ -112,32 +112,7 @@ static void ddramc_reg_config(struct ddramc_register *ddramc_config)
 				| AT91C_DDRC2_UNAL_SUPPORTED);
 
 
-#if defined(CONFIG_BUS_SPEED_133MHZ)
-
-	ddramc_config->rtr = 0x208;
-
-	/* One clock cycle @ 133 MHz = 7.5 ns */
-	ddramc_config->t0pr = (AT91C_DDRC2_TRAS_(6)
-			| AT91C_DDRC2_TRCD_(2)
-			| AT91C_DDRC2_TWR_(2)
-			| AT91C_DDRC2_TRC_(8)
-			| AT91C_DDRC2_TRP_(2)
-			| AT91C_DDRC2_TRRD_(2)
-			| AT91C_DDRC2_TWTR_(2)
-			| AT91C_DDRC2_TMRD_(2));
-
-	ddramc_config->t1pr = (AT91C_DDRC2_TXP_(2)
-			| AT91C_DDRC2_TXSRD_(200)
-			| AT91C_DDRC2_TXSNR_(19)
-			| AT91C_DDRC2_TRFC_(17));
-
-	ddramc_config->t2pr = (AT91C_DDRC2_TFAW_(5)
-			| AT91C_DDRC2_TRTP_(2)
-			| AT91C_DDRC2_TRPA_(2)
-			| AT91C_DDRC2_TXARDS_(2)
-			| AT91C_DDRC2_TXARD_(8));
-
-#elif defined(CONFIG_BUS_SPEED_148MHZ)
+#if defined(CONFIG_BUS_SPEED_148MHZ)
 
 	ddramc_config->rtr = 0x243;
 
@@ -228,7 +203,7 @@ static void ddramc_init(void)
 	pmc_enable_system_clock(AT91C_PMC_DDR);
 
 	/* configure Shift Sampling Point of Data */
-#if defined(CONFIG_BUS_SPEED_133MHZ) || defined(CONFIG_BUS_SPEED_148MHZ)
+#if defined(CONFIG_BUS_SPEED_148MHZ)
 	reg = AT91C_MPDDRC_RD_DATA_PATH_NO_SHIFT;
 #else
 	reg = AT91C_MPDDRC_RD_DATA_PATH_ONE_CYCLES;
@@ -242,7 +217,7 @@ static void ddramc_init(void)
 	reg &= ~AT91C_MPDDRC_CALCODEP;
 	reg &= ~AT91C_MPDDRC_CALCODEN;
 	reg |= AT91C_MPDDRC_RDIV_DDR2_RZQ_50;
-#if defined(CONFIG_BUS_SPEED_133MHZ) || defined(CONFIG_BUS_SPEED_148MHZ)
+#if defined(CONFIG_BUS_SPEED_148MHZ)
 	reg |= AT91C_MPDDRC_TZQIO_4;	/* @ 133 & 148 MHz */
 #else
 	reg |= AT91C_MPDDRC_TZQIO_5;	/* @ 170 & 176 MHz */
diff --git a/board/sama5d4ek/sama5d4ek.h b/board/sama5d4ek/sama5d4ek.h
index 46754c5..682a73f 100644
--- a/board/sama5d4ek/sama5d4ek.h
+++ b/board/sama5d4ek/sama5d4ek.h
@@ -38,33 +38,7 @@
 
 #define BOARD_MAINOSC		12000000
 
-#if defined(CONFIG_CPU_CLK_396MHZ)
-/* PCK: 396M, MCK: 132M */
-#define BOARD_PCK		((unsigned long)((BOARD_MAINOSC * 66) / 2))
-#define BOARD_MCK		((unsigned long)((BOARD_MAINOSC * 66) / 2 / 3))
-
-#define BOARD_CKGR_PLLA		(AT91C_CKGR_SRCA | AT91C_CKGR_OUTA_0)
-#define BOARD_PLLACOUNT		(AT91C_CKGR_PLLACOUNT && (0x3F << 8))
-#define BOARD_MULA		((AT91C_CKGR_MULA << 2) & (65 << 18))
-#define BOARD_DIVA		(AT91C_CKGR_DIVA & 1)
-
-/* Master Clock Register */
-#define BOARD_PRESCALER_MAIN_CLOCK	(AT91C_PMC_PLLADIV2_2 | \
-					AT91C_PMC_MDIV_3 | \
-					AT91C_PMC_CSS_MAIN_CLK)
-
-#define BOARD_PRESCALER_PLLA		(AT91C_PMC_PLLADIV2_2 | \
-					AT91C_PMC_MDIV_3 | \
-					AT91C_PMC_CSS_PLLA_CLK)
-
-#define BOARD_H32MX			(AT91C_PMC_H32MXDIV_H32MXDIV2 \
-					| AT91C_PMC_PLLADIV2_2 \
-					| AT91C_PMC_MDIV_3 \
-					| AT91C_PMC_CSS_PLLA_CLK)
-
-#define MASTER_CLOCK		132000000
-
-#elif defined(CONFIG_CPU_CLK_444MHZ)
+#if defined(CONFIG_CPU_CLK_444MHZ)
 
 /* PCK: 444M, MCK: 148M */
 #define BOARD_PLLA_MULA		73
-- 
1.9.1

